pyrdp/mitm/X224MITM.py

```diff
@@ -117,19 +117,27 @@ def onConnectionConfirm(self, pdu: X224ConnectionConfirmPDU):
         parser = NegotiationResponseParser()
         response = parser.parse(pdu.payload)
         if isinstance(response, NegotiationFailurePDU):
-            if response.failureCode == NegotiationFailureCode.HYBRID_REQUIRED_BY_SERVER and self.state.canRedirect():
-                self.log.info("The server forces the use of NLA. Using redirection host: %(redirectionHost)s:%(redirectionPort)d", {
-                    "redirectionHost": self.state.config.redirectionHost,
-                    "redirectionPort": self.state.config.redirectionPort
-                })
-
-                # Disconnect from current server
-                self.disconnector()
-
-                # Use redirection host and replay sequence starting from the connection request
-                self.state.useRedirectionHost()
-                self.onConnectionRequest(self.originalConnectionRequest)
-                return
+            if response.failureCode == NegotiationFailureCode.HYBRID_REQUIRED_BY_SERVER:
+                if self.state.canRedirect():
+                    self.log.info("The server forces the use of NLA. Using redirection host: %(redirectionHost)s:%(redirectionPort)d", {
+                        "redirectionHost": self.state.config.redirectionHost,
+                        "redirectionPort": self.state.config.redirectionPort
+                    })
+
+                    # Disconnect from current server
+                    self.disconnector()
+
+                    # Use redirection host and replay sequence starting from the connection request
+                    self.state.useRedirectionHost()
+                    self.onConnectionRequest(self.originalConnectionRequest)
+                    return
+                else:
+                    # Respond with a RDP negotiation answer to proceed and catch NTLM
+                    self.log.info("Server requires CredSSP. Closing connection with server and attempting to catch NTML hashes.")
+                    payload = parser.write(NegotiationResponsePDU(NegotiationType.TYPE_RDP_NEG_RSP, 0x00, NegotiationProtocols.CRED_SSP))
+                    self.state.ntlmCatch = True
+
+                    self.disconnector()
```

Comment:
In this case we don't want to return because, unlike the clause above, we are not redirecting the connection but continuing with the SSL connection established on line 149.